x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/Security.java

```diff
@@ -628,6 +636,61 @@ protected Clock getClock() {
         return components;
     }
 
+    private void generateElasticPassword(SecurityIndexManager.State previousState, SecurityIndexManager.State currentState) {
+        if (previousState.equals(SecurityIndexManager.State.UNRECOVERED_STATE)
+            && currentState.equals(SecurityIndexManager.State.UNRECOVERED_STATE) == false
+            && securityIndex.get().indexExists() == false) {
+
+            final SecureString elasticPassword = new SecureString(generatePassword(20));
+            nativeUsersStoreReference.get()
+                .createReservedUser(
+                    ElasticUser.NAME,
+                    elasticPassword.getChars(),
+                    ActionListener.wrap(
+                        r -> {
+                            logger.info("");
+                            logger.info("-----------------------------------------------------------------");
+                            logger.info("");
+                            logger.info("");
+                            logger.info("");
+                            logger.info("Password for the elastic user is: ");
+                            logger.info(elasticPassword);
+                            logger.info("");
+                            logger.info("");
+                            logger.info("Please note this down as it will not be shown again.");
+                            logger.info("");
+                            logger.info("You can use 'bin/elasticsearch-reset-elastic-password' at any time");
+                            logger.info("in order to reset the password for the elastic user.");
+                            logger.info("");
+                            logger.info("");
+                            logger.info("");
+                            logger.info("-----------------------------------------------------------------");
+                            logger.info("");
```

Comment:
I think the listener (if it's a real object) should remove itself at this point. We don't need to call it ever again.